packages/openapi3/src/openapi.ts

```diff
@@ -488,7 +488,7 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
   function getOpenAPIStatuscode(response: HttpOperationResponse): string {
     switch (response.statusCode) {
       case "*":
-        return "default";
+        return "4XX";
```

Comment:
I guess that's the part that is tricky. We are now converting a statusCode being intepreted in the http library as `any` to be an `4xx` which feels like a wrong interpretation.

Reply:
should that change then not be decided at the http layer?

Reply:
should that change then not be decided at the http layer?

Reply:
HTTP has no concept of "4XX" -- only numbered status codes.

Reply:
yeah the library doesn't but it could, we do accept a union of status code. My point is here we are making an interpretation of how a service return the data from the openapi library but in my opinion it should be something the http library dictate. Here we could endup with a generated openapi3 document that doesn't match what another emitter do.

Reply:
I think the HTTP library should be unopinionated. It should implement the RFCs and leave the rest to whatever builds on top of it. 

And here I am simply taking the opinion originally expressed in the OpenAPI3 emitter -- that `@error` maps to "default" -- and changing it to a different and IMHO better opinion to map `@error` to "4XX". The MS Graph team also want "4XX" rather than "default" for their error response, though they aren't using our OpenAPI 3 emitter to do it.

Reply:
I think the distinction is that "default" choice was compatible with "don't know" while "4xx" choice is not. Whether the service only returns 4xx in this case is a matter of fact that every emitter should be able to agree on. I think we do need more expressiveness in the HTTP library to make that possible.

Reply:
And about the part where you think the http library shouldn't be opinionated. It is already extremely opinionated about what it consider the body, responses etc. I would find it weird the draw the line here.

I could understand the want for having the http library just provide the decorators and let another library deal with all the interpretation but also at the same time, why not make another library with same decorator names if you are not going to respect the way they were designed.

Reply:
I was hoping we could avoid having a design discussion on this one but I guess not. I'll move this PR to "draft" for now and we can pick it back up after we've resolved some of these basic issues in a design meeting.